### ISSUE TITLE
Docs suggestion: brief description of how instructor works

### ISSUE THREAD

**Issue by mbforbes** [instructor-ai/instructor#1351]
Hi there, I was reading the readme and the docs, but I couldn't find a concise explanation for how instructor actually works!

I was evaluating whether to use instructor, and wanted to know whether it used constrained decoding, prompting, tool APIs, or something else.

I think it'd be really helpful to have 1–2 sentences in the readme and docs that says something like:

> Here's how instructor gets LLMs to produce responses that follow the provided schema:
> - If an LLM exposes a structured endpoint, we use that
> - Otherwise, if an LLM exposes a tool call endpoint, we use that
> - Otherwise, we fall back to a [prompt](https://github.com/instructor-ai/instructor/blob/4a39ad3199e1829df82e2f159cc30f9b6d40bc96/instructor/process_response.py#L375-L382) to ask the model to output a response of the provided schema. If it fails, we retry N times
> - We don't implement constrained decoding for LLMs that produce logits, but may implement that in the future

(I don't know if the above is accurate!)

As a user of the library, knowing these details are really helpful to understand whether the schema is guaranteed by the LLM in a single call, or whether it's best-effort. Linking to the prompt would be helpful too.

Many thanks for your consideration, and thank you for making and open-sourcing this library!

**Is your feature request related to a problem? Please describe.**
Understanding instructor's approach.

**Describe the solution you'd like**
Brief documentation update

**Describe alternatives you've considered**
Reading the code to figure it out (it's what I did!)

**Additional context**
n/a

**Comment by psharma28** [comment]
In my experience, I noticed that the process kept defaulting back to the prompt to give a structured output, which resulted in a Pydantic errors. I realized this approach is not what I wanted, so I decided to move away from using the instructor.

I opened up this issue, but didn't hear back anything:  #1341

**Comment by ivanleomk** [comment]
I wrote something here at https://ivanleo.com/blog/how-does-instructor-work which should help clear things out!
